src/Analyzers/CSharp/Tests/UseImplicitObjectCreation/CSharpUseImplicitObjectCreationTests.cs

```diff
@@ -0,0 +1,250 @@
+            }.RunAsync();
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsUseImplicitObjectCreation)]
+        public async Task TestWithObjectInitializer()
+        {
+            await new VerifyCS.Test
+            {
+                TestCode = @"
+class C
+{
+    C c = new [|C|]() { };
+}",
+                FixedCode = @"
+class C
+{
+    C c = new() { };
+}",
+                LanguageVersion = CodeAnalysis.CSharp.LanguageVersion.CSharp9,
+            }.RunAsync();
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsUseImplicitObjectCreation)]
+        public async Task TestWithObjectInitializerWithoutArguments()
+        {
+            await new VerifyCS.Test
+            {
+                TestCode = @"
+class C
+{
+    C c = new [|C|] { };
+}",
+                FixedCode = @"
+class C
+{
+    C c = new() { };
+}",
+                LanguageVersion = CodeAnalysis.CSharp.LanguageVersion.CSharp9,
+            }.RunAsync();
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsUseImplicitObjectCreation)]
+        public async Task TestWithTriviaAfterNew()
+        {
+            await new VerifyCS.Test
+            {
+                TestCode = @"
+class C
+{
+    C c = new /*x*/ [|C|]();
+}",
+                FixedCode = @"
+class C
+{
+    C c = new /*x*/ ();
+}",
+                LanguageVersion = CodeAnalysis.CSharp.LanguageVersion.CSharp9,
+            }.RunAsync();
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsUseImplicitObjectCreation)]
+        public async Task TestNotWithDifferentTypes()
+        {
+            await new VerifyCS.Test
+            {
+                TestCode = @"
+class C
+{
+    object c = new C();
+}",
+                LanguageVersion = CodeAnalysis.CSharp.LanguageVersion.CSharp9,
+            }.RunAsync();
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsUseImplicitObjectCreation)]
+        public async Task TestNotWithErrorTypes()
+        {
+            await new VerifyCS.Test
+            {
+                TestState = {
+                    Sources =
+                    {
+                        @"
+class C
+{
+    E c = new E();
+}"
+                    },
+                    ExpectedDiagnostics =
+                    {
+                        // /0/Test0.cs(4,5): error CS0246: The type or namespace name 'E' could not be found (are you missing a using directive or an assembly reference?)
+                        DiagnosticResult.CompilerError("CS0246").WithSpan(4, 5, 4, 6).WithArguments("E"),
+                        // /0/Test0.cs(4,15): error CS0246: The type or namespace name 'E' could not be found (are you missing a using directive or an assembly reference?)
+                        DiagnosticResult.CompilerError("CS0246").WithSpan(4, 15, 4, 16).WithArguments("E"),
+                    }
```

Comment:
💡 I tend to prefer `WithLocation(0)` to `WithSpan`, and then mark location in source with markup syntax `{|#0:code|}`.